src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -187,6 +180,24 @@ def register_options(cls, register):
 
   def register_extra_products_from_contexts(self, targets, compile_contexts):
     super(RscCompile, self).register_extra_products_from_contexts(targets, compile_contexts)
+
+    def cp_entry_for(maybe_file):
+      # Capture a snapshot for the file if it exists
+      if os.path.exists(maybe_file):
+        def capture_or_read_digest(scheduler, filename):
```

Comment:
It's not clear that this needs to be a nested method.

Reply:
Fair. I was originally planning on pulling it up, but I didn't end up needing more usages.